transactionparser/src/orderbookchanges.js

```diff
@@ -0,0 +1,82 @@
+var _ = require('lodash');
+var utils = require('./utils');
+var BigNumber = require('bignumber.js');
+
+/**
+ * Computes the complete list of every Offer that changed in the ledger
+ * as a result of the given transaction.
+ *
+ *  @param {Object} metadata
+ *
+ */
+function parseOrderbookChanges(metadata) {
+  var orderbookChanges = utils.normalizeNodes(metadata).map(function(node) {
+    if (node.entryType === 'Offer') {
+      return parseOrderbookChange(node);
+    }
+  });
+
+  return _.compact(orderbookChanges);
+}
+
+function parseOrderbookChange(node) {
+  var diffType = node.diffType;
+  var status = parseOrderStatus(node);
+
+  return {
+    account: node.finalFields.Account,
+    taker_pays: parseChangeAmount(node, 'TakerPays'),
+    taker_gets: parseChangeAmount(node, 'TakerGets'),
+    sequence: node.finalFields.Sequence,
+    status: status
+  };
+
+  function parseChangeAmount(node,type) {
+    var previousAmount;
+    var finalAmount;
+    var changeAmount;
+    var finalValue;
+    var prevValue;
```

Comment:
Sure, but let's not treat it like gospel. If there is a good reason to justify the rule, then I'm all for it, but I would prefer not to sacrifice code quality for the sake of following a rule that was potentially made under false assumptions.